src/test/shared/defaultAwsContext.test.ts

```diff
@@ -159,31 +208,27 @@ describe('DefaultAwsContext', () => {
         assert.strictEqual(persistedRegions![0], testRegion1Value)
     })
 
-    it('fires event on profile change', async () => {
+    it('fires event on credentials change', async () => {
```

Comment:
Does this test pass if the event doesn't fire? If so, can you force a failure after the Promise if resolve() isn't called?

Reply:
If the event doesn't fire, the test fails with a timeout after two seconds, with a message along the lines of 

```
Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```